docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+# Overview
+VM expose a different, and partially overlapping, set of metrics with
+respect to containers. This document explain the differences and how
+those metrics are made available to the cluster, to be consumed using
+the standard kubernetes facilities.
+
+## libvirt metrics
+[libvirt](https://libvirt.org/) is the virtualization framework that
+kubevirt uses; in order to describe the VM metrics in kubevirt, the
+first step is to inspect what libvirt provides, and how it collect the
+data.
+
+In a nutshell, libvirt uses both the QEMU configuration and the standard
+facilites of the linux kernel to do the resource limitation and
+accounting. As prime example, libvirt uses cgroups, to do resource
+accounting for memory, cpu usage, I/O. Thus, we can consider this subset
+of metrics to fully overlap with what monitoring system (e.g.
+[cAdvisor](https://github.com/google/cadvisor)) already provides.
+Let's call this set `system metrics`.
```

Comment:
We need to be careful here.

cgroups are used in both cases, tho there are different processes in the livirt cgroups and in the kubelet cgoups.

Not sure if it matters.

Reply:
Not sure either, but the concept I was try to convey is that the underlying monitoring mechanism is the same, so we can expect the same metric to be delivered one way or the other; in other words, there is nothing that libvirt reports that cAdvisor can't report as well -and vice versa.
Anyway I'll keep this point in mind and will check in the future.